src/Command/CheckAllSchemaTemplatesDefaultTypeCommand.php

```diff
@@ -0,0 +1,200 @@
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class CheckAllSchemaTemplatesDefaultTypeCommand extends Command
+{
+    private const TYPE_MAP = [
+        "null" => "null",
+        "boolean" => "boolean",
+        "integer" => "int",
+        "string" => "string",
+        "double" => "double",
+        "array" => "array",
+    ];
+
+    /**
+     * @return void
+     */
+    protected function configure(): void
+    {
+        $this
+            ->setName('kafka-schema-registry:check:template:default:type:all')
+            ->setDescription('Checks for default value type for all schema templates in folder')
+            ->setHelp('Checks for default value type for all schema templates in folder')
+            ->addArgument(
+                'schemaTemplateDirectory',
+                InputArgument::REQUIRED,
+                'Path to avro schema template directory'
+            );
+    }
+
+    /**
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     * @return integer
+     * @throws RequestException
+     */
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        /** @var string $directory */
+        $directory = $input->getArgument('schemaTemplateDirectory');
+        $avroFiles = SchemaFileHelper::getAvroFiles($directory);
+
+        $io = new SymfonyStyle($input, $output);
+
+        $failed = [];
+
+        if (false === $this->checkSchemas($avroFiles, $failed)) {
+            $io->error('Following schema templates have invalid default value types:');
+            $io->listing($failed);
+
+            return 1;
+        }
+
+        $io->success('All schema templates have valid default value types');
+
+        return 0;
+    }
+
+    /**
+     * @param array<string, mixed> $avroFiles
+     * @param array<string, mixed> $failed
+     * @return boolean
+     */
+    private function checkSchemas(array $avroFiles, array &$failed = []): bool
+    {
+        $failed = [];
+
+        foreach ($avroFiles as $schemaName => $avroFile) {
+
+            /** @var string $localSchema */
+            $localSchema = file_get_contents($avroFile);
+
+            $invalidFields = $this->checkDefaultType($localSchema);
+
+            if (count($invalidFields)) {
+                foreach ($invalidFields as $invalidField) {
+                    $failed[] = $invalidField;
+                }
+            }
+        }
+
+        return 0 === count($failed);
+    }
+
+    /**
+     * @param string $localSchema
+     * @return array<string, mixed>
+     */
+    private function checkDefaultType(string $localSchema): array
+    {
+        $decodedSchema = json_decode($localSchema);
+        if (!property_exists($decodedSchema, 'fields')) {
+            return [];
+        }
+
+        $result = [
+            'found' => [],
+            'default' => [],
+        ];
```

Comment:
i think we can actually simplify this, since in a template we cannot have overlapping names, you can use the name as index and if you have found it, you can unset that index, so all remaining will be the ones which we didn't find, makes it a bit more readable imho